src/misceval.js

```diff
@@ -360,63 +339,65 @@ Sk.misceval.richCompareBool = function (v, w, op, canSuspend) {
         }
     }
 
-
     // handle identity and membership comparisons
     if (op === "Is") {
-        if (v instanceof Sk.builtin.int_ && w instanceof Sk.builtin.int_) {
-            return v.numberCompare(w) === 0;
-        } else if (v instanceof Sk.builtin.float_ && w instanceof Sk.builtin.float_) {
-            return v.numberCompare(w) === 0;
-        } else if (v instanceof Sk.builtin.lng && w instanceof Sk.builtin.lng) {
-            return v.longCompare(w) === 0;
+        if (v_type === w_type) {
+            if (v === w) {
+                return true;
+            } else if (v_type === Sk.builtin.float_) {
+                return v.v === w.v;
+            } else if (v_type === Sk.builtin.int_) {
+                return v.v === w.v;
+            } else if (v_type === Sk.builtin.lng) {
+                return v.longCompare(w) === 0;
+            }
```

Comment:
What is the argument for breaking the abstraction here and comparing the ".v" values instead of using numberCompare? (Same for "IsNot")

Reply:
As you said. I figured it was before interned was implemented/working and I can't see this line as being necessary.

Reply:
NumberCompare checks if other instanceof int_ which is redundant since we know already. So I didn't see the abstraction as beneficial with float or int here.

Reply:
Makes sense.